### PR TITLE
Socint 228 use updated survey event

### PR DESCRIPTION
**Pull request by PeterBochel**
See original CR for details on changes: https://collaborate2.ons.gov.uk/jira/browse/SOCINT-228

### CODE REVIEW

**Review by rpcatling — CHANGES_REQUESTED** [review]
a few minor things

**Review by spdiaz1000 — APPROVED** [review]

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      this.caseRepo = caseRepo;
+      this.collExRepo = collExRepo;
+      this.productRepo = productRepo;
+      this.ccBeanMapper = ccBeanMapper;
+    }
+
+    public void deleteAll() {
+      caseRepo.deleteAll();
+      collExRepo.deleteAll();
+      productRepo.deleteAll();
```
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java:122]: Test fails if the other deletes are not there.
It fails with a constraint violation exception.

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    private CaseRepository caseRepo;
+    private CollectionExerciseRepository collExRepo;
+    private ProductRepository productRepo;
+    private SurveyRepository repo;
+    private CCSvcBeanMapper ccBeanMapper;
+
+    public SurveyTransactionalOps(
+        CaseRepository caseRepo,
+        SurveyRepository repo,
+        ProductRepository productRepo,
```
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java:109]: Needed to allow test to run. See below.

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+public class SurveyUpdateRepositoryIT extends PostgresTestBase {
+
+  @Autowired private SurveyTransactionalOps txOps;
+
+  @BeforeEach
+  public void setup() throws Exception {
+    txOps.deleteAll();
+
+    // Load a random survey, to make that there is no inteference between the test survey
```
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java:36]: Typo fixed.

File: src/main/resources/db/migration/V11__ProductsForSurvey.sql [hunk 4]
```diff
@@ -0,0 +1,16 @@
+-- add a table to hold products for a survey.
+
+CREATE TABLE product (
+  id UUID NOT NULL,
```
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/main/resources/db/migration/V11__ProductsForSurvey.sql:4]: It is using what has gone before. 
It seems to be only V5 and V9 which use 'id UUID PRIMARY KEY'.
I can see the merits in this but went with the majority

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/ProductRepository.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.repository.db;
+
+import java.util.UUID;
+import org.springframework.data.jpa.repository.JpaRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Product;
+
+public interface ProductRepository extends JpaRepository<Product, UUID> {}
```
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/ProductRepository.java:7]: It's used by the IT test.

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java:122]: This is not happened when I tried it

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java:109]: not when I tried it

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/ProductRepository.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (rpcatling) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/ProductRepository.java:7]: not when I tried it

File: src/main/resources/db/migration/V11__ProductsForSurvey.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (rpcatling) [reply src/main/resources/db/migration/V11__ProductsForSurvey.sql:4]: OK the majority were pretty much all done my me, but I thought this was the best and most succinct.

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java:109]: Now deleted.

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/ProductRepository.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/ProductRepository.java:7]: Now deleted.

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/repository/db/SurveyUpdateRepositoryIT.java:122]: As you say, it works if the 'productRepo' delete is removed.
Now removed :)

**Review by rpcatling — APPROVED** [review]

**Pull request merged**